s3-handler/request.go

```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+	"net/url"
+	"regexp"
+	"strings"
+)
+
+func redactRequest(request string, redactPattern *regexp.Regexp) string {
+	matches := redactPattern.FindAllSubmatchIndex([]byte(request), -1)
+	for _, match := range matches {
+		for i := 1; i < len(match)/2; i++ {
+			start := match[i*2]
+			end := match[i*2+1]
+			request = request[:start] + strings.Repeat("x", end-start) + request[end:]
```

Comment:
I suggest we use `*` instead of `x` to align with the collector's redaction processor. Also, in order to obfuscate the length of the redacted value, for more privacy, the redaction processor uses exactly `****` for any value redacted.